src/internet_identity/src/activity_stats/event_stats.rs

```diff
@@ -0,0 +1,200 @@
+use crate::activity_stats::event_stats::event_aggregations::AGGREGATIONS;
+use crate::ii_domain::IIDomain;
+use crate::state::storage_borrow_mut;
+use crate::storage::Storage;
+use crate::DAY_NS;
+use ic_cdk::api::time;
+use ic_stable_structures::storable::Bound;
+use ic_stable_structures::{Memory, StableBTreeMap, Storable};
+use internet_identity_interface::internet_identity::types::{FrontendHostname, Timestamp};
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+/// This module defines the aggregations over the events.
+mod event_aggregations;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Deserialize, Serialize, Clone, Eq, PartialEq, Debug)]
+pub struct EventData {
+    pub event: Event,
+}
+
+#[derive(Deserialize, Serialize, Clone, Eq, PartialEq, Debug)]
+pub enum Event {
+    PrepareDelegation(PrepareDelegationEvent),
+}
+
+#[derive(Deserialize, Serialize, Clone, Eq, PartialEq, Debug)]
+pub struct PrepareDelegationEvent {
+    pub ii_domain: Option<IIDomain>,
+    pub frontend: FrontendHostname,
+    pub session_duration_ns: u64,
+}
+
+impl Storable for EventData {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(serde_cbor::to_vec(&self).unwrap())
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        serde_cbor::from_slice(&bytes).unwrap()
+    }
+
+    const BOUND: Bound = Bound::Unbounded;
+}
+
+struct WeightedAggregation {
+    key: String,
```

Comment:
Does it make sense to add a comment of how this key looks like and how it's built?